### PR TITLE
Build assembly files with AMBuild.

### PR DESCRIPTION
**Pull request by dvander**
This isn't the prettiest thing but it's better than having these awful object files in the tree. Yasm must be installed on the system. (We could alternately require nasm which would remove the codesize hack).

r?'ing DS for build stuff and Arkshine to see if this would disrupt AMX Mod X developers.

### CODE REVIEW

**Comment by dvander** [comment]
Actually, I think it's reasonable to require nasm. It's in macports/apt and we can add it to checkout-deps.sh.

**Comment by In-line** [comment]
– Hm. So we have 19 build systems.
– Yes, but no one has features that I need.
– Okey, now we have 20 build systems.

**Comment by rsKliPPy** [comment]
> – Hm. So we have 19 build systems.
> – Yes, but no one has features that I need.
> – Okey, now we have 20 build systems.

This change was long overdue actually. To be honest it should have been this way from the very beginning.

Some Wikis should probably be updated as a consequence of this change.

File: AMBuildScript [hunk 1]
```diff
@@ -369,11 +395,42 @@ class AMXXConfig(object):
     binary = context.compiler.Program(name)
     return self.AddVersioning(binary)
 
+  def AddAssembly(self, context, binary, input_file, output_file, includes=[], extra_argv=[]):
+    if builder.target_platform == 'windows':
+      obj_type = 'win32'
+    elif builder.target_platform == 'linux':
+      obj_type = 'elf'
```
Comment (rsKliPPy) [inline AMBuildScript]: Maybe have it as `elf32` as well to match the other two, or remove `32` from the other two. Is there a reason for this one to be different?

**Comment by Arkshine** [comment]
@TheDS If possible, even if it looks good, I would prefer that you say something since david r?'d you.

**Pull request merged**